### PR TITLE
CGAL: Min-max errors on windows

### LINKED ISSUE #4665: Protect CGAL code against min/max macros defined in `<windows.h>`
_Please use the following template to help us solving your issue._

## Issue Details

CGAL 5.0.2-1 appears to introduce a compilation bug on Windows, detected in AppVeyor:

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\include\vcruntime.h(245,47): note: expanded from macro '_CRT_DEPRECATE_TEXT'
#define _CRT_DEPRECATE_TEXT(_Text) __declspec(deprecated(_Text))
                                              ^
In file included from ..\tests\Ergodic_moves_3_test.cpp:11:
In file included from ..\include\Ergodic_moves_3.hpp:14:
In file included from ..\include/Manifold.hpp:14:
In file included from ..\include/Geometry.hpp:19:
In file included from ..\include/Foliated_triangulation.hpp:26:
In file included from C:\Tools\vcpkg\installed\x64-windows\include\CGAL/Exact_predicates_inexact_constructions_kernel.h:20:
In file included from C:\Tools\vcpkg\installed\x64-windows\include\CGAL/Simple_cartesian.h:20:
In file included from C:\Tools\vcpkg\installed\x64-windows\include\CGAL/Cartesian/Cartesian_base.h:22:
In file included from C:\Tools\vcpkg\installed\x64-windows\include\CGAL/Kernel/global_functions.h:25:
C:\Tools\vcpkg\installed\x64-windows\include\CGAL/Kernel/hash_functions.h(60,53): error: too few arguments provided to function-like macro invocation
  std::size_t result = hash_value(iso_rectangle.min());
 
https://ci.appveyor.com/project/acgetchell/cdt-plusplus#L3520

During a recent build:

https://github.com/acgetchell/CDT-plusplus/commit/1a8e0ad29df8b5d6831bb39c558f5fe9440562a1

Note the previous build did not have this problem:

https://ci.appveyor.com/project/acgetchell/cdt-plusplus/builds/31998819
                                                  ^
## Source Code

I have a minimal project here:

https://github.com/acgetchell/CDT-test/

And the AppVeyor build shows the same error here:

https://ci.appveyor.com/project/acgetchell/cdt-test#L4260

## Environment

* Operating system (Windows, 64 bits):
* Compiler: Visual Studio 2017
* Release or debug mode: Release
* Specific flags used (if any): https://github.com/acgetchell/CDT-test/blob/master/cmake/StandardProjectSettings.cmake
* CGAL version: 5.0.2 (5.0.2-1 in `vcpkg`)
* Boost version: 1.72.0 and 1.72.0-1
* Other libraries versions if used (Eigen, TBB, etc.): Eigen 3.3.7-4, TBB 2020_U1-1

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

Fix the errors due to the inclusion of the windows.h.

## Release Management

* Issue(s) solved (if any): fix #4665

### CODE REVIEW

File: Classification/include/CGAL/Classification/compressed_float.h [hunk 1]
```diff
@@ -43,12 +43,12 @@ typedef unsigned char compressed_float;
 inline compressed_float compress_float (const float& f, const float& min = 0.f, const float& max = 1.f)
 {
   return static_cast<compressed_float>
-    (float(std::numeric_limits<compressed_float>::max()) * (f - min) / (max - min));
+    (float((std::numeric_limits<compressed_float>::max)()) * (f - min) / (max - min));
```
Comment (sloriot) [inline Classification/include/CGAL/Classification/compressed_float.h:46]: what about the `min` variable?

File: Classification/include/CGAL/Classification/compressed_float.h [hunk 2]
```diff
@@ -43,12 +43,12 @@ typedef unsigned char compressed_float;
...
 }
 
 inline float decompress_float (const compressed_float& t, const float& min = 0.f, const float& max = 1.f)
 {
-  return ((max - min) * (t / float(std::numeric_limits<compressed_float>::max())) + min);
+  return ((max - min) * (t / float((std::numeric_limits<compressed_float>::max)())) + min);
```
Comment (sloriot) [inline Classification/include/CGAL/Classification/compressed_float.h:51]: same question here

**Comment by lrineau** [comment]
There should be versions of this PR for 4.14.x and 5.0.x. #4665 is an important regression on Windows.

**Comment by gdamiand** [comment]
@maxGimeno Please do not modify CMap, GMap and Surface mesh topology in this pr to avoid conflicts with https://github.com/CGAL/cgal/pull/4602 (I already corrected some min/max in #4602).

**Comment by lrineau** [comment]
> @maxGimeno Please do not modify CMap, GMap and Surface mesh topology in this pr to avoid conflicts with #4602 (I already corrected some min/max in #4602).

I disagree: #4602 is for CGAL-5.2 and later, but this PR is a bug-fix for 4.14.x and later, to fix a regression.

**Comment by gdamiand** [comment]
> > @maxGimeno Please do not modify CMap, GMap and Surface mesh topology in this pr to avoid conflicts with #4602 (I already corrected some min/max in #4602).
> 
> I disagree: #4602 is for CGAL-5.2 and later, but this PR is a bug-fix for 4.14.x and later, to fix a regression.

Maybe for 5.1 ??

**Comment by gdamiand** [comment]
> > @maxGimeno Please do not modify CMap, GMap and Surface mesh topology in this pr to avoid conflicts with #4602 (I already corrected some min/max in #4602).
> 
> I disagree: #4602 is for CGAL-5.2 and later, but this PR is a bug-fix for 4.14.x and later, to fix a regression.

More precisely: ok for a bug fix for 4.14 and 5.0.

For master and for CMap, GMap, LCC, and Surface mesh topology, we can delay a little bit the modification and wait to see if #4602 will be integrated in 5.1 or not.

This will avoid to do the same work several time, and will avoid to solve conflicts. I think we can try to coordinate a little bit our work...

**Comment by maxGimeno** [comment]
There.

**Comment by gdamiand** [comment]
> There.

Thanks @maxGimeno !

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-133.shtml

**Pull request merged**